projects/step-core/src/lib/shared/function-type-form-component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormGroup, ValidatorFn } from '@angular/forms';
+import { Subject, takeUntil } from 'rxjs';
+import { CustomComponent } from '../modules/custom-registeries/custom-registries.module';
+import { FunctionTypeFormComponentContext } from './function-type-form-component-context.interface';
+
+@Component({
+  template: '',
+})
+export abstract class FunctionTypeFormComponent<T extends FormGroup> implements CustomComponent, OnInit, OnDestroy {
+  private readonly terminator$ = new Subject<void>();
+
+  protected abstract formGroup: T;
+  protected abstract formGroupValidator: ValidatorFn;
+
+  context!: FunctionTypeFormComponentContext;
+
+  ngOnInit(): void {
+    this.context!.formGroup.addValidators(this.formGroupValidator);
```

Comment:
Custom subform management looks complicated.
This could be achieved by adding/removing subform dynamically to the parent form group.
In that case validation and validity update will work out of the box.
Probably the setValueToForm / setValueToModel could be also simplified.

Additional ticket will be created to improve this mechanism.